python/ray/_private/services.py

```diff
@@ -1337,7 +1337,7 @@ def start_api_server(
         ray.experimental.internal_kv._initialize_internal_kv(gcs_client)
         dashboard_url = None
         dashboard_returncode = None
-        for _ in range(200):
+        for _ in range(3000):
```

Comment:
while we're at it, can we make this an actual timeout instead of this haphazard approximation?

would also suggest making it an env var just to have an escape hatch to address this issue in the future

Reply:
> can we make this an actual timeout instead of this haphazard approximation?

Do you mean we record the start time and compare it with latest time?

> would also suggest making it an env var just to have an escape hatch to address this issue in the future

That env var shouldn't be considered as public API that we need to maintain compatibility right? Ideally I want to remove timeout here completely but just have the end-to-end `ray start` timeout.

Reply:
> Do you mean we record the start time and compare it with latest time?

Yes otherwise this loop can take arbitrarily long (for example if one of the RPCs is slow or has internal retries).

> That env var shouldn't be considered as public API that we need to maintain compatibility right?

Yea just want an escape hatch private API that we can use just in case we need to adjust it w/o users waiting for a release or needing to patch